test/unittest/unittest_nntrainer_tensor.cpp

```diff
@@ -250,6 +250,52 @@ TEST(nntrainer_Tensor, Tensor_05_p) {
   }
 }
 
+TEST(nntrainer_Tensor, Tensor_06_p) {
+  int status = ML_ERROR_NONE;
+  std::vector<std::vector<std::vector<uint8_t>>> in = {{{0, 1}, {2, 3}},
+                                                       {{4, 5}, {6, 7}},
+                                                       {{8, 9}, {10, 11}},
+                                                       {{12, 13}, {14, 15}}};
+
+  nntrainer::Tensor tensor = nntrainer::Tensor(
+    in, {nntrainer::Tformat::NCHW, nntrainer::Tdatatype::UINT8});
+  ASSERT_NE(nullptr, tensor.getData<uint8_t>());
+
+  for (size_t b = 0; b < tensor.batch(); ++b) {
+    for (size_t c = 0; c < tensor.channel(); ++c) {
+      for (size_t h = 0; h < tensor.height(); ++h) {
+        for (size_t w = 0; w < tensor.width(); ++w) {
+          size_t idx = tensor.getIndex(b, c, h, w);
+          ASSERT_EQ(idx, tensor.getValue<uint8_t>(idx));
+        }
+      }
+    }
+  }
+}
+
+TEST(nntrainer_Tensor, Tensor_07_p) {
+  int status = ML_ERROR_NONE;
+  std::vector<std::vector<std::vector<uint32_t>>> in = {{{0, 1}, {2, 3}},
+                                                        {{4, 5}, {6, 7}},
+                                                        {{8, 9}, {10, 11}},
+                                                        {{12, 13}, {14, 15}}};
+
+  nntrainer::Tensor tensor = nntrainer::Tensor(
+    in, {nntrainer::Tformat::NCHW, nntrainer::Tdatatype::UINT32});
+  ASSERT_NE(nullptr, tensor.getData<uint32_t>());
+
+  for (size_t b = 0; b < tensor.batch(); ++b) {
+    for (size_t c = 0; c < tensor.channel(); ++c) {
+      for (size_t h = 0; h < tensor.height(); ++h) {
+        for (size_t w = 0; w < tensor.width(); ++w) {
+          size_t idx = tensor.getIndex(b, c, h, w);
+          ASSERT_EQ(idx, tensor.getValue<uint32_t>(idx));
+        }
+      }
+    }
+  }
+}
+
```

Comment:
one negative test case that comes to my mind is comparing UInt16Tensor t1 and UInt32Tensor t2 with the same size and values and checking for ASSERT_NE(t1, t2).